adaptation.lua

```diff
@@ -0,0 +1,110 @@
+-- Original opening comment before lua adaptation
+-- View or set cavern adaptation levels
+-- based on removebadthoughts.rb
+-- Rewritten by TBSTeun using OpenAI GPT from adaptation.rb
+
+local function print_color(color, s)
+    dfhack.color(color)
+    dfhack.print(s)
+    dfhack.color(COLOR_RESET)
+end
+
+local function usage(s)
+    if s then
+        print(s)
+    end
+    print(dfhack.script_help())
+end
+
+local function set_adaptation_value(unit, v)
+    local num_set = 0
+
+    if not dfhack.units.isCitizen(unit) or unit.flags2.killed then
+        return num_set
+    end
+    local trait_found = false
+    for _, t in ipairs(unit.status.misc_traits) do
+        if t.id == df.misc_trait_type.CaveAdapt then
+            if mode == "show" then
+                    print_color(COLOR_RESET, "Unit " .. unit.id .. " (" .. dfhack.TranslateName(dfhack.units.getVisibleName(unit)) .. ") has an adaptation of ")
+                    if t.value <= 399999 then
+                        print_color(COLOR_GREEN, t.value .. "\n")
+                    elseif t.value <= 599999 then
+                        print_color(COLOR_YELLOW, t.value .. "\n")
+                    else
+                        print_color(COLOR_RED, t.value .. "\n")
+                    end
+            elseif mode == "set" then
+                print("Unit " .. unit.id .. " (" .. dfhack.TranslateName(dfhack.units.getVisibleName(unit)).. ") changed from " .. t.value .. " to " .. v)
+                t.value = v
+                num_set = num_set + 1
+            end
+            trait_found = true
+            break
+        end
+    end
+    if not trait_found then
+        if mode == "show" then
+            print_color(COLOR_RESET, "Unit " .. unit.id .. " (" .. dfhack.TranslateName(dfhack.units.getVisibleName(unit)) .. ") has an adaptation of ")
+            print_color(COLOR_GREEN, "0\n")
+        elseif mode == "set" then
+            local new_trait = df.unit_misc_trait:new()
+            new_trait.id = df.misc_trait_type.CaveAdapt
+            new_trait.value = v
+            num_set = num_set + 1
+            table.insert(unit.status.misc_traits, new_trait)
+            print("Unit " .. unit.id .. " (" .. dfhack.TranslateName(dfhack.units.getVisibleName(unit)) .. ") changed from 0 to " .. v)
+        end
+    end
+
+    return num_set
+end
+
+local args = {...}
+
+local mode = args[1] or "help"
+local who = args[2]
+local value = args[3]
+
+if mode == "help" then
+    usage()
+elseif mode ~= "show" and mode ~= "set" then
+    usage("Invalid mode '" .. mode .. "': must be either 'show' or 'set'")
```

Comment:
while `..` concatenation does work fine, be aware that it is often easier to read when you use `string:format`. This code could be written as:
```lua
usage(('Invalid mode "%s": must be either "show" or "set"'):format(mode))
```

Reply:
note also that while double and single quotes are interchangeable, the Lua code in DFHack prefers using single quotes for strings. It's fine to keep this PR as it is as far as quote style, but it's something to remember for future code.

Reply:
I feel like using function calls over operators is what is more confusing because there is more information to read. Consider the '+' operator for example, what feels more confusing? 1+1 or add(1, 1)?
That being said, I feel like this is a larger discussion than for pull request comments. I believe consistency is key in making a codebase approachable for new developers and if using string.format is the standard in our coding practices I'll comply and make the change later today. Taking that into account, I will also make the same change on line 39 even though you said leave it in as a reader exercise to keep the codebase approachable for new developers.